canvas/base_shadow.go

```diff
@@ -0,0 +1,101 @@
+package canvas
+
+import (
+	"image/color"
+
+	"fyne.io/fyne/v2"
+)
+
+// ShadowType defines the type of shadow to render.
+type ShadowType int
+
+const (
+	DropShadow ShadowType = iota
+	BoxShadow
+)
+
+// Shadowable defines an interface for objects that can render a shadow.
+// It provides methods to calculate the scaled size and position including shadow paddings,
+// retrieve the shadow paddings, and obtain the content size and position excluding the shadow.
+// Content refers to the area without shadow, while the shadow is rendered outside this area.
+// The canvas object size is reduced to include the shadow within the allocated frame.
+//
+// Since: 2.7
+type Shadowable interface {
+	// SizeAndPositionWithShadow returns the total size and position needed to render the shadow around the content
+	// with the input size. Total size is larger than the input size.
+	SizeAndPositionWithShadow(size fyne.Size) (fyne.Size, fyne.Position)
+	// ShadowPaddings returns the paddings (left, top, right, bottom) for the shadow.
+	ShadowPaddings() [4]float32
```

Comment:
This seems like the only one of the interface methods that adds (rather than works around geometry complications).

However I wonder if ShadowOffset and ShadowDepth really are all we need to draw shadows around objects - all these calculations could be removed and made a rendering only complexity.

Reply:
Thanks. As noted on the recent review now that it is reduced to padding calculations it seems not needed at all...